src/main/scala/scala/tools/refactoring/util/SourceHelpers.scala

```diff
@@ -0,0 +1,44 @@
+package scala.tools.refactoring.util
+
+import scala.annotation.tailrec
+import scala.math.min
+
+object SourceHelpers {
+  def isRangeWithin(text: String, selection: SourceWithSelection): Boolean = {
+    if (selection.length > text.length || selection.source.length < text.length) {
+      false
+    } else {
```

Comment:
why is this else block not just `selection.source.slice(selection.start, selection.end).contains(text)`?